sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -665,13 +705,12 @@ pi_result piDeviceRetain(pi_device Device) {
 
 pi_result piDeviceRelease(pi_device Device) {
   assert(Device);
-
   // TODO: OpenCL says root-device ref-count remains unchanged (1),
   // but when would we free the device's data?
   if (--(Device->RefCount) == 0) {
-    // Destroy the command list used for initializations
-    ZE_CALL(zeCommandListDestroy(Device->ZeCommandListInit));
```

Comment:
Isn't this a memory leak? When are these going to be destroyed eventually?

Reply:
Yes, that's what I mentioned in the previous review. 
Did I misunderstand your previous request?
First you asked "why do we need to invalidate the cache?".
I said "it is not valid to reuse the cached pi_devices after we call DestroyCommandList."
You actually asked me to remove this call, and I actually said it is going to be a memory leak.
You said it would be terminated when the program process ends.
So, I agreed with you and made the change as you requested.

Reply:
So, the new code does not invalidate anything, but sets the invalidated to true, why?

Reply:
We just set the CacheInvalidated flag to indicate that the cache is not valid and when there is a new request for piDevicesGet() we re-initialize the cached pi_devices.

Reply:
What makes the cache invalid? Please add more comments to code, so it clear form the source

Reply:
As soon as there is a call to piDeviceRelease(), the entire cache is invalided.
I will add more comments in the code.

Reply:
Done